includes/class-amp-theme-support.php

```diff
@@ -1497,6 +1497,57 @@ protected static function has_dependency( WP_Dependencies $dependencies, $curren
 		return false;
 	}
 
+	/**
+	 * Check if a handle is exclusively a dependency of another handle.
+	 *
+	 * For example, check if dashicons is being added exclusively because it is a dependency of admin-bar, as opposed
+	 * to being added because it was directly enqueued by a theme or a dependency of some other style.
+	 *
+	 * @since 1.4.2
+	 *
+	 * @param WP_Dependencies $dependencies      Dependencies.
+	 * @param string          $dependency_handle Dependency handle.
+	 * @param string          $dependent_handle  Dependent handle.
+	 * @return bool Whether the $handle is exclusively a handle of the $exclusive_dependency handle.
+	 */
+	protected static function is_exclusively_dependent( WP_Dependencies $dependencies, $dependency_handle, $dependent_handle ) {
+
+		// If a dependency handle is the same as the dependent handle, then this self-referential relationship is exclusive.
```

Comment:
I included this condition as somewhat of an edge case test since if someone has `jquery` as a dependency of `jquery` this actually causes WordPress to self-destruct:

> `PHP Fatal error:  Allowed memory size of 1073741824 bytes exhausted (tried to allocate 20480 bytes) in /wp-includes/class.wp-dependencies.php on line 151`

This is due to `WP_Dependencies::all_deps()` not catching infinite loops: https://github.com/WordPress/wordpress-develop/blob/3b25bc43628a3b0c033248aef260581277880a32/src/wp-includes/class.wp-dependencies.php#L132-L133

The code path won't ever be traversed in how this code is used because `admin-bar` will never a dependency of `admin-bar`. 

So we might as well remove this condition.